sdk/cosmosdb/cosmos/src/common/helper.ts

```diff
@@ -277,7 +278,24 @@ export function validateResourceId(resourceId: string): boolean {
 
   // if resource id contains illegal characters throw an error
   if (illegalResourceIdCharacters.test(resourceId)) {
-    throw new Error("Illegal characters ['/', '\\'] cannot be used in resourceId");
+    throw new Error("Illegal characters ['/', '\\', '#', '?'] cannot be used in resourceId");
```

Comment:
Should we use 'Resource ID' in the error as it's the standard

Reply:
Resolved in next iteration